AxonDeepSeg/segment.py

```diff
@@ -143,6 +143,24 @@ def segment_images(
 
     path_images = [convert_path(p) for p in path_images]
     path_model = convert_path(path_model)
+
+    available_models = ads.get_existing_models_list()
+    if available_models is None:
+        available_models = []
+    if path_model.stem not in available_models:
+            try:
+                print('Model not found, attempting to download')
+                # Call download models from the AxonDeepSeg/download_model.py module
+
+                import AxonDeepSeg.download_model as download_model
+                download_model.download_model()
```

Comment:
Tiny tiny tiny nitpick, but why do we call `download_model()` here but `main()` in the plugin?

They seem to have the same functionality (default values of 'generalist', 'light', and './models/{model_name}'). But, there's some duplication here, so I wonder if there's a risk of a minor code change causing the two different methods to become subtly different. (e.g. https://github.com/axondeepseg/axondeepseg/pull/836)

Could it be beneficial to remove the default arguments of `download_model`, and only store the defaults in `main()`, then standardize on `main()` becoming the primary method of calling this function?

It's fine to save this for a follow-up issue of course! It's not entirely relevant for this PR... :sweat_smile: